core/finality-grandpa/primitives/src/lib.rs

```diff
@@ -46,6 +46,10 @@ pub type AuthoritySignature = app::Signature;
 /// The `ConsensusEngineId` of GRANDPA.
 pub const GRANDPA_ENGINE_ID: ConsensusEngineId = *b"FRNK";
 
+/// The storage key for the current set of weighted Grandpa authorities.
+/// The value stored is an encoded AuthorityList.
+pub const GRANDPA_AUTHORITIES_KEY: &'static [u8] = b":grandpa_authorities";
```

Comment:
One thing I'd keep an eye out for (that the runtime API approach made handling-of easier) is upgradeability.

We might want to upgrade GRANDPA to use BLS keys at some point. 

We could also have a storage item `:grandpa_protocol_version` which holds the protocol version key. We'd need 2 merkle branches, but it would allow us to change the key type later on without issues.

Reply:
I don't see how the runtime API would make this sort of upgrade any easier? Are keys already encoded with the key type? If so, there's no difference, I'd think, and if not, then the `GrandpaApi_grandpa_authorities` return type would have to change anyway?

Reply:
the runtime API has an implicit version, that's why

Reply:
Or we could store the "authorities version" alongside the authorities in the same key. So the value is an encoded `(u8, AuthorityList)`. That way it's still just one merkle proof.

Reply:
sure. that will require a more adaptive `Decode` implementation later on, but it should be fine.

Reply:
Something like 8ee07038?